File: test_binary_search.py
from binary_search import binary_search

if __name__ == '__main__':
    assert binary_search([1], 1)
    assert binary_search([1,2,3], 2)
    assert binary_search([1,2,3,4], 4)
    assert not binary_search([], 1)
    assert not binary_search([2], 1)
    assert not binary_search([2, 3, 4], 1)
    assert not binary_search([2, 3, 4], 5)
    assert not binary_search([1, 3, 4, 5], 2)

File: binary_search/cpp/TestBinarySearch.cpp
#include "BinarySearch.h"

#include <cassert>
#include <iostream>

int main(int argc, char* argv[])
{
	std::vector<int> v = {1};
	assert(binarySearch(v, 1));

	v = {1, 2, 3};
	assert(binarySearch(v, 1));
	assert(binarySearch(v, 2));
	assert(binarySearch(v, 3));

	v = {1, 2, 3, 4};
	assert(binarySearch(v, 4));

	v.clear();
	assert(!binarySearch(v, 1));

	v = {2};
	assert(!binarySearch(v, 1));

	v = {1, 2, 4, 6};
	assert(!binarySearch(v, 0));
	assert(!binarySearch(v, 5));
	assert(!binarySearch(v, 3));

	return 0;
}
